src/services/document-service.js

```diff
@@ -208,33 +208,51 @@ class DocumentService {
 
       lock = await this.documentLockStore.takeLock(documentKey);
 
-      const allRevisions = await this._getAllDocumentRevisionsByKey(documentKey);
+      const revisionsBeforeDelete = await this._getAllDocumentRevisionsByKey(documentKey);
 
-      const revisionsToUpdate = [];
+      const revisionsAfterDelete = [];
+      const revisionsToUpdateInDb = [];
 
-      for (const revision of allRevisions) {
-        for (const section of revision.sections) {
+      for (const originalRevision of revisionsBeforeDelete) {
+        let revisionHasChanged = false;
+        for (const section of originalRevision.sections) {
```

Comment:
This seems to work as well:
```
      for (const originalRevision of revisionsBeforeDelete) {
        let updatedRevision = originalRevision;
        for (const section of originalRevision.sections) {
          if (section.key === sectionKey && !section.deletedOn
             && (section.revision === sectionRevision || deleteAllRevisions)) {
            section.deletedOn = now;
            section.deletedBy = userId;
            section.deletedBecause = reason;
            section.content = null;
            updatedRevision = this._buildDocumentRevision({
              data: originalRevision,
              revisionId: originalRevision._id,
              documentKey: originalRevision.key,
              userId: originalRevision.createdBy,
              order: originalRevision.order,
              restoredFrom: originalRevision.restoredFrom,
              sections: originalRevision.sections
            });
            revisionsToUpdateInDb.push(updatedRevision);
            break;

          }
        }

        revisionsAfterDelete.push(updatedRevision);
      }

Reply:
Thanks, I've changed it (I didn't use the `break` though, which I think is easy to overlook).